### PR TITLE
refactor: LPVSScanossDetectService refactoring and unit test fix

### PR DESCRIPTION
**Pull request by tiokim**
# Pull Request

## Description

The `checkLicenses` has been refactored due to long method.
The unit tests, `testWithNullHeadCommitSHA` and `testWithNullHeadCommitSHADB`, have been fixed to read the existing test file.

## Type of change

- [x] Code cleanup/refactoring

## Testing

This PR was verified by the unit test.

**Test Configuration**:
* Java: v17
* LPVS Release: v1.x.x

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] My code meets the required code coverage for lines (90% and above)
- [ ] My code meets the required code coverage for branches (80% and above)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by o-kopysov — COMMENTED** [review]
Please check my comments. Thanks!

File: src/main/java/com/lpvs/service/scan/scanner/LPVSScanossDetectService.java [hunk 1]
```diff
@@ -261,6 +181,96 @@ public List<LPVSFile> checkLicenses(LPVSQueue webhookConfig) {
         return detectedFiles;
     }
 
+    private Set<LPVSLicense> buildLicenseSet(ScanossJsonStructure object) {
```
[the remark this answers is not in the document]
Comment (tiokim) [reply src/main/java/com/lpvs/service/scan/scanner/LPVSScanossDetectService.java:191]: Javadoc comments generated by GenAI are added. Please let me know if I have anything to revise!

File: src/main/java/com/lpvs/service/scan/scanner/LPVSScanossDetectService.java [hunk 2]
```diff
@@ -147,107 +147,27 @@ public List<LPVSFile> checkLicenses(LPVSQueue webhookConfig) {
...
-                                                + File.separator
-                                                + LPVSPayloadUtil.getRepositoryName(webhookConfig)
-                                                + File.separator
-                                                + webhookConfig.getHeadCommitSHA()
-                                                + ".json"));
-            }
+            Reader reader = getReader(webhookConfig);
+
             // convert JSON file to map
             Map<String, ArrayList<Object>> map =
```
[the remark this answers is not in the document]
Comment (tiokim) [reply src/main/java/com/lpvs/service/scan/scanner/LPVSScanossDetectService.java:152]: Done. Thanks for your precise review!

File: src/main/java/com/lpvs/service/scan/scanner/LPVSScanossDetectService.java [hunk 3]
```diff
@@ -261,6 +181,96 @@ public List<LPVSFile> checkLicenses(LPVSQueue webhookConfig) {
...
+                        licenseService.addLicenseConflict(incompatibleLicense, license.name);
+                    }
+                }
+            }
+        }
+        return licenses;
+    }
+
+    private ScanossJsonStructure getScanossJsonStructure(Gson gson, String content, LPVSFile file) {
+        ScanossJsonStructure object = gson.fromJson(content, ScanossJsonStructure.class);
```
[the remark this answers is not in the document]
Comment (tiokim) [reply src/main/java/com/lpvs/service/scan/scanner/LPVSScanossDetectService.java]: Done. Thanks for your precise review!

**Review by o-kopysov — APPROVED** [review]
Thank you for the code refactoring!

**Review by tdrozdovsky — APPROVED** [review]

**Pull request merged**